controllers/exclude_instances.go

```diff
@@ -70,6 +70,19 @@ func (e ExcludeInstances) Reconcile(r *FoundationDBClusterReconciler, context ct
 	}
 
 	if len(addresses) > 0 {
+		// Block excludes on missing processes not marked for removal
+		missingProcesses := make([]string, 0)
+		for _, processGroupStatus := range cluster.Status.ProcessGroups {
+			missingTime := processGroupStatus.GetConditionTime(fdbtypes.MissingProcesses)
+			if missingTime != nil && !processGroupStatus.Remove {
```

Comment:
We probably also want to check for `MissingPod` https://github.com/FoundationDB/fdb-kubernetes-operator/blob/master/api/v1beta1/foundationdbcluster_types.go#L607 with the current design of the `ProcessGroups` it's a little bit harder but I think `MissingPod` and `MissingProcesses` should be fine. We might want to consider to create a new `ProcessGroup` with one of these conditions to be sure we don't run into a situation where we created the new `ProcessGroup` but the controller didn't set any condition on it.

Reply:
+1 to Johannes's suggestions. I think it makes sense to add the `MissingPod` condition when we create a new process group in AddProcessGroups, since at the time we add the process group that we don't have a corresponding pod yet.